### PR TITLE
[READY] Ian's Birthday Holiday

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

I've added the date Ian got age added to him (if you have ian's actual birthday or something, definitely let me know!). On Ian's birthday, the hop room turns into a birthday party. What does this entail?

- [x] Balloons are scattered throughout the room
- [x] Ian has a birthday cake on the table!
- [x] ...and a corgi balloon on his bed!
- [x] There is confetti everywhere
- [x] Hop gets an Ian shirt
~There are decorations on the wall~

Admin wise, there is a version that only deletes itself after the proc for the birthdayization happens, so admins can spontaneously turn any room into a festive party.

Code wise, there are now random colored balloons that come in 4 colors. We also have a corgi balloon, but I need to finish that. I'll put it in Ian's bed.

## Why It's Good For The Game

Ian is the center of more hop gimmicks than you think, and every time it happens it warms my cold, dead heart. I really like the idea that the hop gives out cake in the hopline on Ian's birthday, or something silly like that.

## Changelog
:cl:
add: Ian's birthday is now a holiday! Make sure to visit the head of personnel to wish his dog a happy birthday!
/:cl:

### CODE REVIEW

**Comment by Detective-Google** [comment]
anyone who dislikes this PR is going to hell

**Comment by optimumtact** [comment]
YOU NEVER HAD A HEART IN THE FIRST PLACE

**Comment by carshalash** [comment]
This is a nice PR

File: code/modules/mapping/mapping_helpers.dm [hunk 1]
```diff
@@ -230,3 +230,56 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+		var/list/dirs_to_balloon = GLOB.alldirs.Copy()
+		while(balloons_left_to_give > 0)
+			if(!dirs_to_balloon.len)
+				for(var/ii in 1 to balloons_left_to_give)
+					new /obj/item/toy/balloon(clusterspot)
+			var/turf/balloonstep = get_step(clusterspot, pick_n_take(dirs_to_balloon))
+			if(isopenturf(balloonstep))
+				new /obj/item/toy/balloon(balloonstep)
+				balloons_left_to_give--
+	//remind me to add confetti and wall decor!
```
Comment (Akrilla) [inline code/modules/mapping/mapping_helpers.dm]: add confetti and wall decor

**Comment by OverlyGeneric** [comment]
god, this is going to be adorable when you get Ian as a puppy when this happens.

also please make the AI Upload have Ian's Birthday Lawset when birthday-ization happens.

**Comment by tralezab** [comment]
I want to leave it to the hops to make the most out of their dog's birthday

**Comment by optimumtact** [comment]
Why did you remove the smoothing code?

**Comment by tralezab** [comment]
I was gonna do some neat stuff with the confetti bunching towards the walls but it really did not look nice at all. I tried turning it into wall decorations near the walls but it looked even worse. I'm still thinking on the best way to do some fun wall stuff (maybe just set up some posters?) but either way smoothing isn't really going to help out

File: code/modules/mapping/mapping_helpers.dm (hunk 1 above)
Comment (tralezab) [reply code/modules/mapping/mapping_helpers.dm]: eh i got one of them seems good

**Comment by tralezab** [comment]
i want someone to look at the mapping helper and tell me if i'm insane or not. the rest should be pristine

File: code/modules/jobs/job_types/head_of_personnel.dm [hunk 2]
```diff
@@ -51,3 +51,9 @@
 		/obj/item/melee/classic_baton/telescopic=1, /obj/item/modular_computer/tablet/preset/advanced = 1)
 
 	chameleon_extras = list(/obj/item/gun/energy/e_gun, /obj/item/stamp/hop)
+
+/datum/outfit/job/hop/pre_equip(mob/living/carbon/human/H)
+	..()
+	if(locate(/datum/holiday/ianbirthday) in SSevents.holidays)
+		H.undershirt = /datum/sprite_accessory/undershirt/ian
+		H.update_body()
```
Comment (optimumtact) [inline code/modules/jobs/job_types/head_of_personnel.dm]: you should not need to do this in the pre_equip

Comment (optimumtact) [reply code/modules/jobs/job_types/head_of_personnel.dm]: honestly I'd prefer that you add undershirt to the available types in the datum and then auto call this during equip if it's necessary to show the undershirt on the body

File: code/modules/mapping/mapping_helpers.dm [hunk 3]
```diff
@@ -230,3 +230,71 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+
+//On Ian's birthday, the hop's office is decorated.
+/obj/effect/mapping_helpers/ianbirthday
+	name = "Ian's Bday Helper"
+	late = TRUE
+	icon_state = "iansbdayhelper"
+	var/balloon_clusters = 2
+	var/delete_after = FALSE
+
+/obj/effect/mapping_helpers/ianbirthday/LateInitialize()
```
Comment (optimumtact) [inline code/modules/mapping/mapping_helpers.dm]: why not just in init? the items will all exist at that piont

File: code/modules/jobs/job_types/head_of_personnel.dm (hunk 2 above)
Comment (tralezab) [reply code/modules/jobs/job_types/head_of_personnel.dm]: sure thing

Comment (tralezab) [reply code/modules/jobs/job_types/head_of_personnel.dm]: added!

File: code/modules/mapping/mapping_helpers.dm (hunk 3 above)
Comment (tralezab) [reply code/modules/mapping/mapping_helpers.dm]: does not work

**Comment by PKPenguin321** [comment]
This is good and all but Ian is definetly going to be killed a disproportionate amount on his birthday due to all the attention on him 😓

**Comment by tralezab** [comment]
i'd be... very disappointed in the hop if they let this happen, you know.

**Comment by PKPenguin321** [comment]
If there even is a HoP...

**Comment by actioninja** [comment]
set your calenders boys

File: code/modules/holiday/holidays.dm [hunk 4]
```diff
@@ -589,3 +589,11 @@ Since Ramadan is an entire month that lasts 29.5 days on average, the start and
 
 /datum/holiday/easter/getStationPrefix()
 	return pick("Fluffy","Bunny","Easter","Egg")
+
+/datum/holiday/ianbirthday
+	name = "Ian's Birthday"
+	begin_day = 15
+	begin_month = MARCH
```
Comment (cacogen) [inline code/modules/holiday/holidays.dm]: Ian's birthday is actually 10 Sep 2011 UTC+0. Here's the commit where errorage added corgis and Ian:
https://github.com/tgstation/tgstation/commit/de7e4f0de0d568cd6e1f0d7bcc3fd34700598acb#diff-9aa879074aa89b230889437f7c112619R61
You can compare tgstation.2.0.8.dmm or mob.dmi with their previous versions to see the absence of Ian on the map and corgi sprites in that file.

Comment (actioninja) [reply code/modules/holiday/holidays.dm]: March 15th was chosen as it's the date where Ian's age was first started to be tracked.

**Pull request merged**